serialize/src/flags.rs

```diff
@@ -91,33 +113,21 @@ impl Flags for SWFlags {
     }
 
     #[inline]
-    fn from_u8(value: u8) -> Self {
+    fn from_u8(value: u8) -> Option<Self> {
         let x_sign = (value >> 7) & 1 == 1;
         let is_infinity = (value >> 6) & 1 == 1;
         match (x_sign, is_infinity) {
-            (_, true) => SWFlags::Infinity,
-            (true, false) => SWFlags::PositiveY,
-            (false, false) => SWFlags::NegativeY,
+            (true, true) => None,
+            (false, true) => Some(SWFlags::Infinity),
+            (true, false) => Some(SWFlags::PositiveY),
+            (false, false) => Some(SWFlags::NegativeY),
         }
     }
-
-    #[inline]
-    fn from_u8_remove_flags(value: &mut u8) -> Self {
-        let flags = Self::from_u8(*value);
-        *value &= 0x3F;
-        flags
-    }
-
-    /// Number of bits required for these flags.
-    #[inline]
-    fn len() -> usize {
-        2
-    }
 }
 
 /// Flags to be encoded into the serialization.
 /// The default flags (empty) should not change the binary representation.
-#[derive(Clone, Copy)]
+#[derive(Clone, Copy, PartialEq, Eq)]
 pub enum EdwardsFlags {
```

Comment:
Can there be comments for how the flags are added to the encoding? (Appears to be using the last bit of the serialization)